tests/queries/test_qs_combinators.py

```diff
@@ -450,6 +459,27 @@ def test_union_in_subquery(self):
             [8, 1],
         )
 
+    @skipUnlessDBFeature("supports_select_intersection")
+    def test_intersection_in_nested_subquery(self):
+        tag = Tag.objects.create(name="tag")
+        note = Note.objects.create(tag=tag)
+        annotation = Annotation.objects.create(tag=tag)
+        tags = Tag.objects.order_by()
+        tags = tags.filter(id=OuterRef("tag_id")).intersection(
+            tags.filter(id=OuterRef(OuterRef("tag_id")))
+        )
```

Comment:
Previously the SQL would be

```sql
SELECT *
FROM "queries_note"
WHERE EXISTS
    (SELECT 1 AS "a"
     FROM "queries_annotation" V0
     INNER JOIN "queries_annotation_notes" V2 ON (V0."id" = V2."annotation_id")
     WHERE (EXISTS
              (SELECT U0."id"
               FROM "queries_tag" U0
               WHERE U0."id" = ("queries_annotation"."tag_id") INTERSECT
                 SELECT U0."id"
                 FROM "queries_tag" U0 WHERE U0."id" = ("queries_note"."tag_id")
               LIMIT 1)
            AND V2."note_id" IN ("queries_note"."id"))
     LIMIT 1)
```

Which would crash because of the `U0."id" = ("queries_annotation"."tag_id")` part.

This was due to the `tags.filter(id=OuterRef("tag_id"))` part resolving to `"queries_annotation"."tag_id"` before `"queries_annotation"` was re-labelled to `V0` when the the `Exists(Annotation...)` annotation was resolved in the `Note.objects.filter` clause.

The changes to `sql.Query.change_aliases` make sure that when this happens the re-label of `{"queries_annotation": "V0"}` percolate all the way to `tags` in order to replace the `WHERE` clause to `U0."id" = ("V0"."tag_id")`.